### PR TITLE
[UI Tests] - Part 2 - Update screens to use same structure

### PR DESCRIPTION
**Pull request by jostnes**
### Description
This is the continuation of https://github.com/wordpress-mobile/WordPress-iOS/pull/21122 where this PR updates all other screens other than login and editor-related screens. 

There are a couple of [flaky tests](https://buildkite.com/automattic/wordpress-ios/builds/15601) that are not introduced by this PR (those are also happening in trunk) and will be taken care of in a separate PR. 

### Testing
CI should be 🟢

### CODE REVIEW

File: WordPress/UITests/Flows/LoginFlow.swift [hunk 1]
```diff
@@ -32,12 +32,4 @@ class LoginFlow {
             .proceedWithValidPassword()
             .continueWithSelectedSite(title: selectedSiteTitle)
     }
-
-    // Login with self-hosted site via Site Address.
-    static func loginIfNeeded(siteUrl: String, username: String, password: String) throws -> TabNavComponent {
```
Comment (jostnes) [inline WordPress/UITests/Flows/LoginFlow.swift:37]: this is no longer used anywhere

File: WordPress/UITestsFoundation/Screens/MySiteScreen.swift [hunk 2]
```diff
@@ -1,126 +1,140 @@
...
-    static let pagesCardMoreButton = "More"
-    static let pagesCardCreatePageButton = "Create another page"
-    // "Activity Log" Card
-    static let activityLogCardId = "dashboard-activity-log-card-frameview"
-    static let activityLogCardHeaderButton = "Recent activity"
-}
-
 /// The home-base screen for an individual site. Used in many of our UI tests.
 public class MySiteScreen: ScreenObject {
-    public let tabBar: TabNavComponent
```
Comment (jostnes) [inline WordPress/UITestsFoundation/Screens/MySiteScreen.swift:37]: removing `tabBar` from individual screens. we can access tab bar separately from the test. adding it on every screen is an extra check for the first element on the screen _and_ the tab bar element which can make the test slow, without adding much value to the test (as we would know if there are issues on the tab bar from the tab bar tests)

File: WordPress/UITestsFoundation/Screens/DomainsScreen.swift [hunk 3]
```diff
@@ -2,17 +2,14 @@ import ScreenObject
...
+    private let siteDomainsNavbarHeaderGetter: (XCUIApplication) -> XCUIElement = {
         $0.staticTexts["Site Domains"]
     }
 
     var siteDomainsNavbarHeader: XCUIElement { siteDomainsNavbarHeaderGetter(app) }
 
-    public init(app: XCUIApplication = XCUIApplication()) throws {
-        tabBar = try TabNavComponent()
-
+    init(app: XCUIApplication = XCUIApplication()) throws {
```
Comment (jostnes) [inline WordPress/UITestsFoundation/Screens/DomainsScreen.swift:12]: if this screen isn't explicitly called from tests, it doesn't have to be `public`

File: WordPress/UITestsFoundation/Screens/Media/MediaPickerAlbumScreen.swift [hunk 4]
```diff
@@ -26,20 +39,10 @@ public class MediaPickerAlbumScreen: ScreenObject {
...
+        selectedActionButton.tap()
     }
 
     public static func isLoaded(app: XCUIApplication = XCUIApplication()) -> Bool {
-        // Check if the media picker is loaded as a component within the editor
-        // and only return true if the media picker is a full screen
-        if app.navigationBars["Azctec Editor Navigation Bar"].exists {
-            return false
-        }
-
```
Comment (jostnes) [inline WordPress/UITestsFoundation/Screens/Media/MediaPickerAlbumScreen.swift:42]: this navigation bar doesn't exist on the app anymore so this condition can be removed

File: WordPress/UITestsFoundation/Screens/MySiteScreen.swift [hunk 5]
```diff
@@ -1,126 +1,140 @@
...
-    static let createButton = "floatingCreateButton"
-    static let ReaderButton = "Reader"
-    static let switchSiteButton = "SwitchSiteButton"
-    static let dashboardButton = "Home"
-    static let segmentedControlMenuButton = "Menu"
-    // "Free To Paid Plans" Card
-    static let freeToPaidPlansCardId = "dashboard-free-to-paid-plans-card-contentview"
-    static let freeToPaidPlansCardHeaderButton = "Free domain with an annual plan"
-    // "Pages" Card
-    static let pagesCardId = "dashboard-pages-card-frameview"
```
Comment (pachlava) [inline WordPress/UITestsFoundation/Screens/MySiteScreen.swift:26]: I'd actually vote for keeping this, `freeToPaidPlansCardId` and `activityLogCardId`.

On the example with `pagesCardId`, after the changes, the `dashboard-pages-card-frameview` string is used four times in getters, and also used for a `pagesCardId` string. 

WDYT?

**Review by pachlava — COMMENTED** [review]
Hey @jostnes 👋 Once again, a massive refactoring that makes things easier / tidier 👍 

I left only one comment for `MySiteScreen`, this is the only reason why I'm not giving an explicit approval.

File: WordPress/UITestsFoundation/Screens/MySiteScreen.swift (hunk 5 above)
Comment (jostnes) [reply WordPress/UITestsFoundation/Screens/MySiteScreen.swift:26]: good eye! yeah looks like quite a bit of repetition there, reverted that in https://github.com/wordpress-mobile/WordPress-iOS/pull/21133/commits/69451c02444646a7760b1e994953d91c5d9ccf69

**Review by pachlava — APPROVED** [review]
Good to go after the most recent changes, @jostnes 🎉 :shipit:

**Pull request merged**